types/firestore.d.ts

```diff
@@ -1065,23 +1065,56 @@ declare namespace FirebaseFirestore {
   }
 
   /**
-   * Sentinel values that can be used when writing document fields with set()
-   * or update().
+   * Sentinel values that can be used when writing document fields with set(),
+   * create() or update().
    */
   export class FieldValue {
     private constructor();
 
     /**
-     * Returns a sentinel used with set() or update() to include a
+     * Returns a sentinel used with set(), create() or update() to include a
      * server-generated timestamp in the written data.
+     *
+     * @return The FieldValue sentinel for use in a call to set(), create() or
+     * update().
      */
     static serverTimestamp(): FieldValue;
 
     /**
-     * Returns a sentinel for use with update() to mark a field for deletion.
+     * Returns a sentinel for use with update() or set() with {merge:true} to
+     * mark a field for deletion.
+     *
+     * @return The FieldValue sentinel for use in a call to set() or update().
      */
     static delete(): FieldValue;
 
+    /**
+     * Returns a special value that can be used with set(), create() or update()
+     * that tells the server to union the given elements with any array value
+     * that already exists on the server. Each specified element that doesn't
+     * already exist in the array will be added to the end. If the field being
+     * modified is not already an array it will be overwritten with an array
+     * containing exactly the specified elements.
+     *
+     * @param elements The elements to union into the array.
+     * @return The FieldValue sentinel for use in a call to set(), create() or
+     * update().
+     */
+    static arrayUnion(...elements: any[]): FieldValue;
```

Comment:
Does it make sense to have a public equivalent of `AnyJS`?

Reply:
Let's talk about this in the API discussion, but I would like to keep this file as similar to the firestore.d.ts file we use for the Web client.